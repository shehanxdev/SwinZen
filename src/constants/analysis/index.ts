export * from './analysis.enum';
export * from './pro-tips-content';
