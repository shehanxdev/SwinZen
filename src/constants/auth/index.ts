export * from './otp-data.enum';
