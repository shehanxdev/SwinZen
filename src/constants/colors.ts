/* eslint-disable @typescript-eslint/no-shadow */
/* eslint-disable @typescript-eslint/no-redeclare */

//NOTE:: Since an enum member cannot have a numeric name prefixed with "Sz"
enum Primary {
  Sz900 = `#070707`,
  Sz800 = `#0C1E0F`,
  Sz700 = `#16501D`,
  Sz600 = `#28642A`,
  Sz500 = `#45734A`,
  Sz400 = `#69AA67`,
  Sz300 = `#8DBE8B`,
  Sz200 = `#C0DBBF`,
  Sz100 = `#F0F7F0`, //NOTE::figma contain two different colors(#F0F7F0, #C0DBBF) with the border. #C0DBBF neeeds to be removed since it does not consider within the colors
}

enum Neutral {
  Sz1000 = `#283E36`, //NOTE::not available within the design system colors
  Sz900 = `#212529`,
  Sz800 = `#343A40`,
  Sz700 = `#495057`,
  Sz600 = `#6C757D`,
  Sz500 = `#ADB5BD`,
  Sz400 = `#CED4DA`,
  Sz300 = `#DEE2E6`,
  Sz200 = `#E9ECEF`,
  Sz100 = `#F8F9FA`, //NOTE::figma contain two different colors(#F8F9FA, #DEE2E6) with the border. #DEE2E6 neeeds to be removed since it does not consider within the colors
  Black = `#000000`,
  White = `#FFFFFF`,
}

enum Secondary {
  Sz900 = `#F65815`,
  Sz800 = `#F87944`,
  Sz700 = `#F99469`,
  Sz600 = `#FAA17C`,
  Sz500 = `#FBBCA1`,
  Sz400 = `#FDD7C7`,
  Sz300 = `#FDE4D9`,
  Sz200 = `#FFF2ED`,
  Sz100 = `#FFE3B0`, //NOTE::figma contain two different colors(##FFE3B0, #FFF9F6) with the border. #FFF9F6 neeeds to be removed since it does not consider within the colors
}

/* Currently this is available within the design system as a Secondary value. Since two different colors cannot exist in there, this has to be added to the design system */
enum Tertiary {
  Sz900 = `#A2FD2F`,
  Sz800 = `#B5FD59`,
  Sz700 = `#C4FE7A`,
  Sz600 = `#CBFE8A`,
  Sz500 = `#DAFEAC`,
  Sz400 = `#E9FFCD`,
  Sz300 = `#F0FEDE`,
  Sz200 = `#F8FFEF`,
  Sz100 = `#DAFEAC`, //NOTE::figma contain two different colors(#DAFEAC, #FBFFF7) with the border. #FBFFF7 neeeds to be removed since it does not consider within the colors
}

enum Transparency {
  Sz100 = `#212529`,
  Sz80 = `#212529cc`,
  Sz64 = `#212529a3`,
  Sz56 = `#2125298f`,
  Sz40 = `#21252966`,
  Sz24 = `#2125293d`,
  Sz16 = `#21252929`,
  Sz8 = `#21252914`,
  Sz4 = `#2125290a`,
  full = `#ffffff00`, //NOTE:: This is full transpapernt color. Not available under the design system.
}

enum Error {
  SzMain = `#E35F5F`, //Not defined within the design system but being used within the screens :(
}

const SwingZenThemePalette = {
  Primary,
  Neutral,
  Secondary,
  Tertiary,
  Transparency,
  Error,
};

export type Color = Primary | Neutral | Secondary | Tertiary | Transparency | Error;

export const Color = {
  ...SwingZenThemePalette,
};
