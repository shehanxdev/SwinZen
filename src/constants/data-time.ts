export const dateFormats = {
  default: 'YYYY-MM-DD',
  calenderHeader: 'MMMM YYYY',
  rangeDatePicker: 'MMM DD',
  scratchedDateAndTime: 'HH:MM, DD/MM/YY',
  singleDatePicker: 'ddd, MMM DD',
  textInput: 'DD / MM / YY',
  extrasMarketHeader: 'dd/MM/YY',
  finalResultsHeader: 'dd/MM/YYYY',
  dayAndDate: 'ddd, DD',
  timestamp: 'DD/MM/YYYY HH:mm:ssPM Z',
  advertisedStartTime: 'ddd, DD MMM YYYY HH:mm PM',
  saveFileTimestamp: 'DD_MM_YYYY_HH_mm_ss',
};
