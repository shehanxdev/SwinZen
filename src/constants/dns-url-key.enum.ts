//TODO::fill other dns url keys
export enum DnsUrlKey {
  Auth0 = 'auth0',
}
