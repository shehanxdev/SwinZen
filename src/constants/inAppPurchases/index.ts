export * from './plan-sku-data';
