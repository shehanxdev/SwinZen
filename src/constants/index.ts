export * from './auth';
export * from './error';
export * from './colors';
export * from './data-time';
export * from './navigation-routes';
export * from './regular-expressions';
export * from './runtime-environment';
export * from './static-info';
export * from './typography';
