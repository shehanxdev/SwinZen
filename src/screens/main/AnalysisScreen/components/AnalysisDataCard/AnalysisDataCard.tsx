import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { szdayjs, tw } from '@sz/config';
import { Color, ScoreType, TextVariant } from '@sz/constants';
import { addPadToNumber } from '@sz/utils';

export interface AnalysisDataCardProps {
  testID?: string;
  score: number;
  observation: ScoreType;
  time: Date;
}

export function AnalysisDataCard({
  testID = 'AnalysisDataCardTestID',
  score,
  observation,
  time,
}: AnalysisDataCardProps) {
  const currentMonth = szdayjs().get('month');
  const actualMonth = szdayjs(new Date(time)).get('month');

  return (
    <View
      testID={testID}
      style={tw`flex-row h-16 mx-2 px-5 bg-Neutral-Sz1000 rounded-2.5 items-center justify-between`}>
      <View style={tw`flex-row items-center gap-7.5`}>
        {/* design shadow and this implemntation has a different, here can't use arbitary values to shadow, added closest tailwind style */}
        <View
          style={tw`w-10 h-10 bg-[${
            score < 5 ? Color.Secondary.Sz900 : Color.Primary.Sz500
          }] items-center justify-center rounded-full shadow-md shadow-Neutral-Black/25`}>
          <Text variant={TextVariant.SubTitle2SemiBold} color={Color.Primary.Sz100}>
            {addPadToNumber(score)}
          </Text>
        </View>
        <Text variant={TextVariant.Body2SemiBold} color={Color.Neutral.White}>
          {observation}
        </Text>
      </View>
      <View style={tw`flex-row pl-7.5 items-center gap-2.5 border-l-2 border-Neutral-Sz100`}>
        <Text variant={TextVariant.Body2Regular} color={Color.Primary.Sz200}>
          {currentMonth === actualMonth + 1 ? 'last month' : szdayjs(time).format('MMMM')}
        </Text>
        <Text variant={TextVariant.SubTitle1} color={Color.Neutral.White}>
          {addPadToNumber(szdayjs(time).date())}
        </Text>
      </View>
    </View>
  );
}
