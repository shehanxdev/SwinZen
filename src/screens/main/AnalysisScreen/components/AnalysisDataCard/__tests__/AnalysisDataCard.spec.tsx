import { render } from '@testing-library/react-native';
import * as React from 'react';

import { Checkpoint } from '@sz/constants';

import { AnalysisDataCard } from './../AnalysisDataCard';

describe('AnalysisData Card Component', () => {
  const testID = 'AnalysisDataCardTestID';
  const mockScore = 10;
  const mockObservation = Checkpoint.OVERALL;
  const mockDate = new Date('2023-05-30');

  const getRenderedComponent = () =>
    render(<AnalysisDataCard testID={testID} score={mockScore} observation={mockObservation} time={mockDate} />);

  beforeEach(() => {
    jest.useFakeTimers().setSystemTime(new Date('2023-05-30'));
  });

  it(`should render AnalysisData Card component correctly`, () => {
    const rendered = getRenderedComponent();
    const renderedTree = rendered.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it('should find the AnalysisData Card component via testID', () => {
    const { getByTestId } = getRenderedComponent();
    const foundAnalysisDataCardComponent = getByTestId(testID);
    expect(foundAnalysisDataCardComponent).toBeTruthy();
  });
});
