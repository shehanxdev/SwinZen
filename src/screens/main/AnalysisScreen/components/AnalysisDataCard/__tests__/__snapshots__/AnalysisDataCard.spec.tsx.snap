// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalysisData Card Component should render AnalysisData Card component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#283E36",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "marginBottom": 7,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 7,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
  testID="AnalysisDataCardTestID"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 30,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#45734A",
          "borderRadius": 9999,
          "elevation": 3,
          "height": 40,
          "justifyContent": "center",
          "shadowColor": "rgba(0, 0, 0, 0.25)",
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.125,
          "shadowRadius": 3,
          "width": 40,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#F0F7F0",
              "lineHeight": 28,
            },
            {
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 24,
              "fontWeight": "600",
              "textAlign": "center",
            },
          ]
        }
      >
        10
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 24,
          },
          {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Overall
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#F8F9FA",
        "borderLeftWidth": 2,
        "flexDirection": "row",
        "gap": 10,
        "paddingLeft": 30,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#C0DBBF",
            "lineHeight": 24,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "center",
          },
        ]
      }
    >
      June
    </Text>
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 40,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 28,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      09
    </Text>
  </View>
</View>
`;
